rfcs/20221024-persistent-cache-api-for-ocl-engine/README.md

```diff
@@ -0,0 +1,169 @@
+# Add Persistent Cache API for OpenCL Engine
+
+## Motivation
+In order to get all required information about GPU device oneDNN has to
+compile a dummy OpenCL kernel to use the compiler binary for querying that
+information.
+The kernel compilation happens during engine creation. In order to mitigate
+the associated overhead oneDNN caches the compiled binaries however, this
+doesn't help when the engine is created for the first time.
+
+OpenVINO reported that first engine creation overhead makes their network
+loading mechanism twice as slow.
+
+There is an option to rely on CL cache (implemented in driver). The main
+drawback of the option is that it requires to modify registry on Windows which
+doesn't work for some OpenVINO customers. Because of that OpenVINO has its own
+caching mechanism that caches serialized OpenCL kernels on disk.
+
+## Proposal
+
+oneDNN already provides persistent cache API for OpenCL runtime that allowed
+OpenVINO to eliminate first primitive creation overhead. OpenVINO can the same
+approach to eliminate first engine creation.
+
+The proposal is to introduce persistent cache API for engine. The API will be
+OpenCL runtime specific therefore it will reside in OpenCL interop namespace.
+
+### C API
+```c
+// dnnl_ocl.h
+
+/// Retrieves a cache blob ID for the OpenCL device.
+///
+/// @warning
+///     This API is intended to be used with
+///     #dnnl_ocl_interop_engine_get_cache_blob() and
+///     #dnnl_ocl_interop_engine_create_from_cache_blob(). The returned cache
+///     blob ID can only be used as an ID of the cache blob returned by
+///     #dnnl_ocl_interop_engine_get_cache_blob().
+///
+/// @note The cache blob ID can be empty (@p size will be 0 and
+///     @p cache_blob_id will be nullptr) if oneDNN doesn't have anything to
+///     put in the cache blob. (#dnnl_ocl_interop_engine_get_cache_blob will
+///     return an empty cache blob).
+///
+/// @param device An OpenCL device.
+/// @param size Size of the cache blob ID in bytes.
+/// @param cache_blob_id Cache blob id of size @p size. If
+///     the @p cache_blob_id is nullptr then the size of the cache blob ID is
+///     returned in @p size.
+/// @returns #dnnl_success on success and a status describing the error
+///     otherwise.
```

Comment:
nit: the sequence of comment is different from argument.

Reply:
Fixed, thanks